### ISSUE TITLE
Errors are not handled correctly

### ISSUE THREAD

**Issue by agrafix** [nikita-volkov/hasql-transaction#6]
I'm currently working on code that looks like this:

```haskell
import Hasql.Transaction

foo :: Transaction BazResult
foo =
    do sql "-- many create table statements here"
       r <- query baz (statement bazQ bazE bazD False)
       pure r
```

and running that using `Tx.transaction Tx.ReadCommitted Tx.Write action`. In the `-- many create table statements here` part I had an error:

```
column \"cird\" referenced in foreign key constraint does not exist
```

however with the code above I got the following error:

```
SessionError (ClientError (Just "another command is already in progress\n"))
```

To get the correct error I had to move the code out of a transaction. Then I get the correct error:

```
SessionError (ResultError (ServerError "42703" "column \"cird\" referenced in foreign key constraint does not exist" Nothing Nothing))
```

**Comment by nikita-volkov** [comment]
Thanks!

I'll look into it.

**Comment by AndrewRademacher** [comment]
This issue is fixed in [here](https://github.com/nikita-volkov/hasql/pull/78), in the underlying `hasql` library.

**Comment by nikita-volkov** [comment]
Please confirm that it's fixed in 0.19.8.2.

**Comment by shinzui** [comment]
I think this issue can be closed.
